src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/get_options_list_control_factory.tsx

```diff
@@ -86,72 +86,48 @@ export const getOptionsListControlFactory = (): DataControlFactory<
       const hideExists = initialState.hideExists;
       const hideSort = initialState.hideSort;
 
-      /** Runtime / component state - none of this is serialized */
-      const searchString$ = new BehaviorSubject<string>('');
-      const searchStringValid$ = new BehaviorSubject<boolean>(true);
-      const requestSize$ = new BehaviorSubject<number>(MIN_OPTIONS_LIST_REQUEST_SIZE);
-
-      const dataLoading$ = new BehaviorSubject<boolean | undefined>(undefined);
-      const availableOptions$ = new BehaviorSubject<OptionsListSuggestions | undefined>(undefined);
-      const invalidSelections$ = new BehaviorSubject<Set<OptionsListSelection>>(new Set());
-      const totalCardinality$ = new BehaviorSubject<number>(0);
+      const temporaryStateManager = initializeTemporayStateManager();
```

Comment:
Why have two different state managers for these states rather than just one? The fact that this state isn't serialized could be defined at the `serializeState` function via picking only the pieces of state which should be serialized. To me that might be more explicit also. 

Additionally, the name `temporaryState` isn't very descriptive - but I can't think of a better name at the moment.

Reply:
Its more than just `serializeState`, its also `anyStateChange$`. If the state in `temporaryState` is combined with another manager, then `anyStateChange$` when temporary state changes - which we do not want.